Sources/Apodini/Utilities/EventLoopFuture.swift

```diff
@@ -0,0 +1,34 @@
+//
+//  File.swift
+//  
+//
+//  Created by Paul Schmiedmayer on 1/12/21.
+//
+
+import NIO
+
+extension EventLoopFuture {
+    /// When the current EventLoopFuture<Value> is fulfilled, run the provided callback, which will provide a new EventLoopFuture.
+    /// The provided callback may optionally `throw`.
+    ///
+    /// This allows you to dynamically dispatch new asynchronous tasks as phases in a longer series of processing steps.
+    /// Note that you can use the results of the current EventLoopFuture<Value> when determining how to dispatch the next operation.
+    ///
+    /// If your callback function throws, the returned `EventLoopFuture` will error.
+    ///
+    /// - parameters:
+    ///     - callback: Function that will receive the value of this EventLoopFuture and return a new EventLoopFuture. The provided `callback` may optionally `throw`.
+    /// - returns: A future that will receive the eventual value.
+    @inlinable
+    public func tryFlatMap<NewValue>(file: StaticString = #file,
+                                     line: UInt = #line,
+                                     _ callback: @escaping (Value) throws -> EventLoopFuture<NewValue>) -> EventLoopFuture<NewValue> {
```

Comment:
i'd call this `transform` iirc that's the terminology used in the swift stdlib

Reply:
I wasn't sure about the naming too. As all `EventLoopFuture` functions are named `map` and `flatMap` and `flatMapThrowing` is weirdly a `map` that can throw. And there is no `mapThrowing` 🤷‍♂️

At fist I didn't want to also name the function `flatMapThrowing`. But looking at the naming problem now maybe it might be better to provide an overload to `flatMapThrowing` and add a `mapThrowing` function to clearly indicate the difference in code between the two?

Reply:
@PSchmiedmayer just out of curiosity what does `#file` / `#line` mean, it this the same as `#function`?

Reply:
@PSchmiedmayer i was talking about the `callback` parameter.

I'd say `tryFlatMap` is fine, if we want to clearly differentiate from `flatMapThrowing`. I'm not sure how i'd feel about overloading the function with different closure return types, although i guess it'd be fine since they both do the same thing, and in most cases the user doesn't actually care if its a map or a flatMap operation, they just want to somehoe process a future into a different future.

Reply:
@hendesi There are a few of these literal expression in Swift. The Language reference provides a good overview over everything that is currently exposed using the `#...` literal expressions: https://docs.swift.org/swift-book/ReferenceManual/Expressions.html#ID389

Reply:
Ah I see, I was missing that your comment referred to `callback`. I took `callback` as it is the same name used in `NIO` across the implementations of `flatMap` and so on.

I will rename `tryFlatMap` to `flatMapThrowing` so it is fully in line with the NIO naming. I agree with your opinion that is is probably not really important for the develop to differentiate between the overloads. I will also change the scope to internal so we don't expose the function for now.